### ISSUE TITLE
`\\\n` in string not recognized as new line

### ISSUE THREAD

**Issue by ABaldwinHunter** [seattlerb/ruby_parser#222]
foo.rb
```
def foo(user)
  puts "hello \
  what is up \
  my dear friend? \
  sup?"
end


def foo(user)
  puts "hello
  what is up
  my dear friend?
  sup?"
end
```

console
```console
$ str = File.read("foo.rb")
$ str
=> "def foo(user)\n  puts \"hello \\\n  what is up \\\n  my dear friend? \\\n  sup?\"\nend\n\n\ndef foo(user)\n  puts \"hello\n  what is up\n  my dear friend?\n  sup?\"\nend\n"

$ RubyParser.new.parse(str)
=> s(:block, s(:defn, :foo, s(:args, :user), s(:call, nil, :puts, s(:str, "hello   what is up   my dear friend?   sup?"))),
s(:defn, :foo, s(:args, :user), s(:call, nil, :puts, s(:str, "hello\n  what is up\n  my dear friend?\n  sup?"))))
```

Suspect that this difference causes tools built on top (like flay) to report inaccurate line numbers. I haven't gotten to work on fix yet - may take a crack at that! Posting issue for now.

cc @maxjacobson who helped hunt down this bug

**Comment by zenspider** [comment]
There's two problems here. 1) "\\\n" in a string is not a newline and 2) The actual newline in there isn't being counted. The first is either ambiguous language in the title or a misunderstanding. The second is now fixed via the test provided in #223.

**Comment by ABaldwinHunter** [comment]
@zenspider 

>  2) The actual newline in there isn't being counted.

Was the main concern I had. Thanks for addressing in https://github.com/seattlerb/ruby_parser/pull/223!

I can go ahead and close this issue now.
